Added some documentation

JacobSherrod opened the pull request:
https://github.com/meilisearch/meilisearch-java/issues/71
This isn't exhaustive but what I was able to gather so far in one evening of looking at the code. Please let me know what I have gotten right and wrong up to this point so I can be more accurate in later commits

eskombro commented on src/main/java/com/meilisearch/sdk/Client.java:
```diff
@@ -42,10 +42,10 @@ public Index createIndex(String uid) throws Exception {
 	 * @param primaryKey The primary key of the documents in that index
 	 * @return Meilisearch API response
-	 * @throws Exception If an error occurs
+	 * @throws Exception if the JSON cannot be parsed correctly
```
What you wrote is true, but it is just a part of it. Imagine you try to create an Index that already exists. This wouldn't be a JSON parsing problem, yet an Exception will be thrown! So in this case I think it may be better to leave something more generic

JacobSherrod replied:
Looking at it the first time I though the Exception was only being thrown from the `gson.fromJson()`. Didn't think about the call to `indexesHandler.getAll()`. Thanks

eskombro commented on src/main/java/com/meilisearch/sdk/Client.java:
```diff
@@ -58,7 +58,7 @@ public Index createIndex(String uid, String primaryKey) throws Exception {
 	 *
 	 * @return List of indexes in the Meilisearch client
-	 * @throws Exception If an error occurs
+	 * @throws Exception if the JSON cannot be parsed correctly
```
Note: same as my previous comment

eskombro commented on src/main/java/com/meilisearch/sdk/Documents.java:
```diff
@@ -15,16 +15,36 @@ protected Documents(Config config) {
+	 * Retrieves the document at the specified uid with the specified identifier
+	 * @param uid Partial path to the requested document
+	 * @param identifier ID of the document
+	 * @return Requested document
```
```suggestion
	 * @return A String containint the requested document
```
Good, but could be interesting to mention the type! 👍

eskombro commented on src/main/java/com/meilisearch/sdk/Documents.java:
```diff
@@ -15,16 +15,36 @@ protected Documents(Config config) {
 
+	/**
+	 * Retrieves the document at the specified uid with the specified identifier
+	 * @param uid Partial path to the requested document
```
I wouldn't mention this as a 'path' but as an Index identifier

eskombro commented on src/main/java/com/meilisearch/sdk/Documents.java:
```diff
@@ -15,16 +15,36 @@ protected Documents(Config config) {
 
+	/**
+	 * Retrieves the documents at the specified uid
+	 * @param uid Partial path to the requested documents
+	 * @return Requested documents
```
Same comments as previous method

eskombro reviewed (commented):
This is exactly what we were expecting and a very good contribution. I think this could improve a lot developer experience for some users regarding this SDK. Thanks a lot! 🎉 

I added a few comments on some details, but what you are doing is great! :)

eskombro commented:
Ping me anytime you want to check any details, or whenever you want an exhaustive review!

curquiza commented:
Hello @JacobSherrod! Thanks for this awesome PR!! That a huuuuuge work 👍 

Did you finish to apply all the changes asked by @eskombro? If yes, we are going to do a final review and then merge your PR.

⚠️ There are git conflicts so you need to fix them before we do any review! If you need help to solve the conflicts, join us on [our Slack](https://slack.meilisearch.com/). It would be easier to discuss. I'm Clémentine on Slack, and I would be happy to guide you.

JacobSherrod commented:
I addressed the comments from earlier. It is probably ready. I meant to look over it and the online documentation over the weekend but ended up having to take care of some stuff. I'll try to take care of the merge conflicts after I finish my FTE work today 👍

curquiza commented:
No rush! Thanks for your involvement @JacobSherrod!

JacobSherrod commented:
@eskombro @curquiza It should be merged and ready

curquiza reviewed (approved):
huge work @JacobSherrod! Thanks a lot! Waiting for the @eskombro review and we will merge! :)

eskombro reviewed (approved):
Thanks a lot for your work @JacobSherrod ! 👍🏽 

Amazing :)

eskombro commented:
bors merge
